### PR TITLE
feat: http error handler

### LINKED ISSUE #148: Support for Http error handler
Currently, if there is an exception on the channel, there is no way for the HTTP app to handle it.
Typically someone might want to log it and probably close the connection after an exception is thrown.

**Example**
```scala
Server.port(PORT) ++       
  Server.app(app) ++
  Server.error(cause => log.throwable(cause).as(true))
```

### PR DESCRIPTION
**Pull request by dinojohn**
fixes #148

### CODE REVIEW

File: zio-http/src/main/scala/zhttp/service/Server.scala [hunk 1]
```diff
@@ -65,7 +69,7 @@ object Server {
       zExec          <- UnsafeChannelExecutor.make[R].toManaged_
       channelFactory <- ZManaged.access[ServerChannelFactory](_.get)
       eventLoopGroup <- ZManaged.access[EventLoopGroup](_.get)
-      httpH           = ServerRequestHandler(zExec, settings.http)
+      httpH           = ServerRequestHandler(zExec, settings.http, settings.errorHandler)
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/Server.scala:72]:
```suggestion
      httpH           = ServerRequestHandler(zExec, settings)
```

File: zio-http/src/main/scala/zhttp/service/server/ServerRequestHandler.scala [hunk 2]
```diff
@@ -91,7 +92,10 @@ final case class ServerRequestHandler[R](
    */
   override def exceptionCaught(ctx: JChannelHandlerContext, cause: Throwable): Unit = {
     if (self.canThrowException(cause)) {
```
Comment (tusharmath) [inline zio-http/src/main/scala/zhttp/service/server/ServerRequestHandler.scala:94]: I think we can deprecate `canThrowException`.

File: zio-http/src/main/scala/zhttp/service/Server.scala (hunk 1 above)
Comment (dinojohn) [reply zio-http/src/main/scala/zhttp/service/Server.scala:72]: settings is a private case class for server object

Comment (tusharmath) [reply zio-http/src/main/scala/zhttp/service/Server.scala]: we can make it `private [zhttp]`

**Review by tusharmath — APPROVED** [review]

**Pull request merged**